database/migrations/0008-state_users_plug_holes.ts

```diff
@@ -0,0 +1,23 @@
+/* MIGRATION TO CLEAN PRODUCTION DATA
+ *
+ * some entries in the state_users table do not have an email.
+ * this is required tho to work with the new environment.
+ * to mitigate 38 out of 50 emails could be restored from
+ * login_users.
+ */
+
+export async function upgrade(queryFn: (query: string, values?: any[]) => Promise<Array<any>>) {
+  // Fill in missing emails from login_users
+  await queryFn(
+    `UPDATE state_users
+     INNER JOIN login_users ON state_users.public_key = login_users.pubkey
+     SET state_users.email = IF(state_users.email = '', login_users.email, state_users.email)
```

Comment:
state_users.email should always be empty since the where clause only take state_users.email = ''.
Would the check not be better like this:
```suggestion
     SET state_users.email = IF(login_users.email != '', login_users.email, state_users.email)
```

Reply:
Good catch!